src/NodaTime.Test/DurationTest.Construction.cs

```diff
@@ -6,16 +6,57 @@
 using NUnit.Framework;
 using System.Numerics;
 using static NodaTime.NodaConstants;
+using System.Linq;
 
 namespace NodaTime.Test
 {
     partial class DurationTest
     {
+        // Test cases for factory methods. In general, we want to check the limits, very small, very large
+        // and medium values, with a mixture of positive and negative, "on and off" day boundaries.
+
+        private static readonly int[] DayCases = {
+            Duration.MinDays, -3000 * 365, -100, -1, 0, 1, 100, 3000 * 365, Duration.MaxDays
+        };
+
+        private static readonly int[] HourCases = {
+            Duration.MinDays * HoursPerDay, -3000 * 365 * HoursPerDay, -100, -48, -1, 0,
+            1, 48, 100, 3000 * 365 * HoursPerDay, (Duration.MaxDays + 1) * HoursPerDay - 1
+        };
+
+        private static readonly long[] MinuteCases = GenerateCases(MinutesPerDay);
+
+        private static readonly long[] SecondCases = GenerateCases(SecondsPerDay);
+
+        private static readonly long[] MillisecondCases = GenerateCases(MillisecondsPerDay);
+
+        // No boundary versions as Int64 doesn't have enough ticks to reach out limits.
```

Comment:
nit: "to reach the limits"? "to reach past the limits"? (not sure what this should say).